src/specfem2d.cpp

```diff
@@ -323,17 +335,50 @@ PYBIND11_MODULE(_core, m) {
            _run
     )pbdoc";
 
-    m.def("_run", &_run, R"pbdoc(
+    m.def("_initialize", &_initialize, R"pbdoc(
+        Initialize SPECFEM++.
+    )pbdoc");
+
+    m.def("_execute", &_execute, R"pbdoc(
         Execute the main SPECFEM++ workflow.
     )pbdoc");
 
+    m.def("_finalize", &_finalize, R"pbdoc(
+        Finalize SPECFEM++.
+    )pbdoc");
+
+    m.attr("_default_file_path") = __default_file__;
+
 #ifdef VERSION_INFO
     m.attr("__version__") = MACRO_STRINGIFY(VERSION_INFO);
 #else
     m.attr("__version__") = "dev";
 #endif
 }
 
-#endif
+#else // SPECFEMPP_BINDING_PYTHON
+
+int main(int argc, char **argv) {
```

Comment:
So, just for me to understand this. We compile specfem2d twice, once further up in the CMakeLists.txt without ENABLE_PYTHON, and then one time with to enable calling the executable directly and then once again with ENABLE_PYTHON to enable binding with python, right?

Reply:
yes, but only specfem2d.cpp is compiled twice